src/Analyzers/Core/CodeFixes/MakeMemberStatic/AbstractMakeMemberStaticCodeFixProvider.cs

```diff
@@ -29,22 +50,90 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
             return Task.CompletedTask;
         }
 
-        protected sealed override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor,
+        private static bool TryGetIndexOfRawKind(SyntaxTokenList modifiers, int rawKind, out int index)
+        {
+            for (var i = 0; i < modifiers.Count; i++)
+            {
+                if (modifiers[i].RawKind == rawKind)
+                {
+                    index = i;
+                    return true;
+                }
+            }
+
+            index = -1;
+            return false;
+        }
+
+        private SyntaxTokenList InsertStaticModifier(SyntaxTokenList modifiers, string preferredOrder)
+        {
+            if (modifiers.Count == 0)
+            {
+                return new SyntaxTokenList(StaticModifier);
+            }
+
+            var order = preferredOrder.Split(',').Select(m => m.Trim()).ToArray();
+            var staticIndex = Array.IndexOf(order, "static");
+
+            if (staticIndex == -1)
+            {
+                return modifiers.Add(StaticModifier);
+            }
+
+            int? modifierBeforeStaticRawKind = null;
+            for (var i = staticIndex - 1; i >= 0; i--)
+            {
+                var rawKind = GetKeywordRawKind(order[i]);
+                if (TryGetIndexOfRawKind(modifiers, rawKind, out var index))
+                {
+                    modifierBeforeStaticRawKind = rawKind;
+                    break;
+                }
+            }
+
+            if (!modifierBeforeStaticRawKind.HasValue)
+            {
+                // When the static modifier is added at the beginning of the modifiers it needs to
+                // get the leading trivia from the previous first modifier
+                return new SyntaxTokenList(StaticModifier.WithLeadingTrivia(modifiers[0].LeadingTrivia),
+                    modifiers[0].WithoutLeadingTrivia()).AddRange(modifiers.Skip(1));
+            }
+
+            var keywords = new Collection<SyntaxToken>();
```

Comment:
```suggestion
            using var _ = ArrayBuilder<SyntaxToken>.GetInstance(out var keywords);
```

Reply:
Changed to array builder with sufficient capacity.